### PR TITLE
fix(AppMain): put dropRectangle in a Loader, active when dragging

### LINKED ISSUE #6023: Review why it is used `dropRectangle` on top of `AppMain.qml`, it interfers with the `Squish` auto testing development
### Description
`AppMain.qml` contains a `dropRectangle` component placed on top of the qml that interfers with the `Squish` auto testing development bc it is not possible to directly pick qml objects to map them to the auto test project since the `dropRectangle` is on top of the view. 

Review why this specific rectangle is on top and if it is another way of redesigning it to be able to pick directly other components of the view from `Squish`.

The interference rectangle:
````
Rectangle {
                id: dropRectangle

                width: parent.width
                height: parent.height
                color: Style.current.transparent
                opacity: 0.8

                states: [
                    State {
                        when: dragTarget.enabled && dragTarget.containsDrag
                        PropertyChanges {
                            target: dropRectangle
                            color: Style.current.background
                        }
                    }
                ]
            }
````

**NOTE:** Workaround done now in `Squish` auto-test development:
- Comment this specific rectangle.
- Select the required objects
- Add objects into the object map file
- Then undo the changes done in `AppMain.qml` file.

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #6023

Puts the dropRectangle in a Loader that is only active when we are dragging

### CODE REVIEW

**Comment by jrainville** [comment]
@noelia-santos can you test if it works and also fixes the squish issues?

File: ui/app/mainui/AppMain.qml [hunk 1]
```diff
@@ -811,24 +811,20 @@ Item {
...
-                ]
+            Loader {
+                active: dragTarget.enabled && dragTarget.containsDrag
+                width: active ? parent.width : 0
+                height: active ? parent.height : 0
+                sourceComponent: Rectangle {
+                    id: dropRectangle
+
+                    width: parent.width
+                    height: parent.height
```
Comment (osmaczko) [inline ui/app/mainui/AppMain.qml:822]:
```suggestion
```
"If the size of the Loader is specified explicitly by setting the width, height or by anchoring, the loaded item will be resized to the size of the Loader."

Comment (jrainville) [reply ui/app/mainui/AppMain.qml]: TIL

**Comment by noeliaSD** [comment]
> @noeliaSD can you test if it works and also fixes the squish issues?

It seems to solve the problem but I'd like @OlufemiAdeOlusile to confirm bc I've tryied to run the current tests and once the tests land into the main page, they keep blocked bc they are not able to get the expected object (but maybe we need to reajust the objects in the object map file) but if we watch the following videos, we can see that with the new changes _Squish_ can get the objects that where in the back:

- PREVIOUS (without Jo changes):

https://user-images.githubusercontent.com/97019400/178954938-544f2e26-fcab-4c02-a01d-2c8e03f3a8b4.mov

- CURRENT (with Jo changes):

https://user-images.githubusercontent.com/97019400/178954952-3603cf09-ef1f-4b16-88ab-c3dc5d601e05.mov

**Review by noeliaSD — APPROVED** [review]
Go for it! :)

**Pull request merged**